k_tree/find_matches.py

```diff
@@ -0,0 +1,20 @@
+"""Return the nodes that match the val."""
+
+from .k_tree import KTree
+
+
+def find_matches(tree, val):
+    """Return the nodes that match the val."""
+    new = KTree()
```

Comment:
Instead of creating a new KTree, it might be easier to store the items in a list.